<?xml version="1.0"?>
<robot name="robotic_arm">

    <material name="Black">
    	<color rgba="0 0 0 1"/>
    </material>
    
    <material name="Blue">
    	<color rgba="0 0 0.8 1"/>
    </material>
    
    <material name="Red">
    	<color rgba="1 0 0 1"/>
    </material>
    
    <material name="Yellow">
    	<color rgba="0 1 0.5 1"/>
    </material>
    
    
    <link name="world"/>
    
    <link name="base_link">
        <visual>
            <geometry>
                <cylinder length="0.2" radius="0.3"/>
            </geometry>
            <material name="Black"/>
            <origin rpy="0 0 0" xyz="0 0 0.025"/>
        </visual>
    </link>
    

    <link name="link_1">
        <visual>
            <geometry>
                <cylinder length="0.5" radius="0.05"/>
            </geometry>
            <material name="Blue"/>
            <origin rpy="0 0 0" xyz="0 0 0.3"/>
        </visual>
    </link>
    

    <link name="link_2">
        <visual>
            <geometry>
                <cylinder length="0.1" radius="0.08"/>
            </geometry>
            <material name="Red"/>
        </visual>
    </link>


    <link name="link_3">
        <visual>
            <geometry>
                <cylinder length="0.4" radius="0.05"/>
            </geometry>
            <material name="Blue"/>
        </visual> 
    </link>
    
    
    <link name="link_4">
        <visual>
            <geometry>
                <cylinder length="0.1" radius="0.06"/>
            </geometry>
            <material name="Red"/>
        </visual>
    </link>


    <link name="link_5">
        <visual>
            <geometry>
                <cylinder length="0.3" radius="0.03"/>
            </geometry>
            <material name="Yellow"/>
        </visual>
    </link>




    <joint name="fixed" type="fixed">
        <parent link="world"/>
        <child link="base_link"/>
    </joint>
    
    <joint name="joint_1" type="continuous">
        <axis xyz="0 0 1"/>
        <parent link="base_link"/>
        <child link="link_1"/>
        <origin rpy="0 0 0" xyz="0.0 0.0 0.05"/>
    </joint>

    <joint name="joint_2" type="continuous">
        <axis xyz="0 0 1"/>
        <parent link="link_1"/>
        <child link="link_2"/>
        <origin rpy="0 1.5708 0" xyz="0.0 -0.005 0.58"/>
    </joint>

    <joint name="joint_3" type="continuous">
    	<axis xyz="0 0 1"/>
        <parent link="link_2"/>
        <child link="link_3"/>
        <origin rpy="1.57 0 0" xyz="0.0 0.2 0 "/>
    </joint>

    <joint name="joint_4" type="continuous">
    	<axis xyz="0 0 1"/>
        <parent link="link_3"/>
        <child link="link_4"/>
        <origin rpy="1.57 0 0" xyz=" 0 0 -0.25"/>
    </joint>

    <joint name="joint_5" type="continuous">
    	<axis xyz="0 0 1"/>
        <parent link="link_4"/>
        <child link="link_5"/>
        <origin rpy="1.57 0 0" xyz="0.0 -0.2 0 "/>
    </joint>

</robot>
